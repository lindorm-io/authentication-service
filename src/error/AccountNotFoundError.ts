import { APIError, HttpStatus } from "@lindorm-io/core";

export class AccountNotFoundError extends APIError {
  constructor(accountId?: string) {
    super("Account not found", {
      debug: { accountId },
      statusCode: HttpStatus.ClientError.BAD_REQUEST,
    });
  }
}
