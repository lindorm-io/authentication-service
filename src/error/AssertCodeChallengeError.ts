import { APIError, HttpStatus } from "@lindorm-io/core";

export class AssertCodeChallengeError extends APIError {
  constructor(challenge: string, verifier: string) {
    super("Invalid Code Challenge", {
      debug: { challenge, verifier },
      statusCode: HttpStatus.ClientError.FORBIDDEN,
    });
  }
}
