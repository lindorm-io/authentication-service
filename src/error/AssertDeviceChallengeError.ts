import { APIError } from "@lindorm-io/errors";
import { HttpStatus } from "@lindorm-io/core";

export class AssertDeviceChallengeError extends APIError {
  constructor(challenge: string, verifier: string) {
    super("Invalid Device Challenge", {
      debug: { challenge, verifier },
      statusCode: HttpStatus.ClientError.FORBIDDEN,
    });
  }
}
