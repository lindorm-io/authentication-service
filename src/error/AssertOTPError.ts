import { APIError } from "@lindorm-io/errors";
import { HttpStatus } from "@lindorm-io/core";

export class AssertOTPError extends APIError {
  constructor() {
    super("Invalid OTP", {
      statusCode: HttpStatus.ClientError.FORBIDDEN,
    });
  }
}
