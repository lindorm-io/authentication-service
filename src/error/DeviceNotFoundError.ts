import { APIError, HttpStatus } from "@lindorm-io/core";

export class DeviceNotFoundError extends APIError {
  constructor(deviceId?: string) {
    super("Device not found", {
      debug: { deviceId },
      statusCode: HttpStatus.ClientError.BAD_REQUEST,
    });
  }
}
