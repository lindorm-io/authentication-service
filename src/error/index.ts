export * from "./AccountNotFoundError";
export * from "./AssertClientSecretError";
export * from "./AssertCodeChallengeError";
export * from "./AssertDeviceChallengeError";
export * from "./AssertOTPError";
export * from "./DeviceNotFoundError";
export * from "./ExpiredSessionError";
export * from "./InvalidAuthorizationError";
export * from "./InvalidAuthorizationTokenError";
export * from "./InvalidClientError";
export * from "./InvalidDeviceError";
export * from "./InvalidExpiryString";
export * from "./InvalidGrantTypeError";
export * from "./InvalidPermissionError";
export * from "./InvalidRefreshTokenError";
export * from "./InvalidScopeError";
export * from "./InvalidSubjectError";
export * from "./LockedAccountError";
export * from "./RejectedClientError";
export * from "./RequestLimitBackOffError";
