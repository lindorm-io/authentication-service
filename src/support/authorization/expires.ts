import { isAfter } from "date-fns";
import { Authorization } from "../../entity";
import { ExpiredAuthorizationError } from "../../error";

export const assertAuthorizationIsNotExpired = (authorization: Authorization): void => {
  if (isAfter(new Date(), new Date(authorization.expires))) {
    throw new ExpiredAuthorizationError(authorization);
  }
};
