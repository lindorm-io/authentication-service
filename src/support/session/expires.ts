import { add, isAfter } from "date-fns";
import { stringToDurationObject } from "@lindorm-io/core";
import { Session } from "../../entity";
import { ExpiredSessionError } from "../../error";

export const getSessionExpires = (expiry: string): Date => {
  return add(Date.now(), stringToDurationObject(expiry));
};

export const assertSessionIsNotExpired = (session: Session): void => {
  if (isAfter(new Date(), new Date(session.expires))) {
    throw new ExpiredSessionError(session);
  }
};
