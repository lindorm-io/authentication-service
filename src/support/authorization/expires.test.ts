import MockDate from "mockdate";
import { ExpiredAuthorizationError } from "../../error";
import { assertAuthorizationIsNotExpired } from "./expires";
import { getTestAuthorization } from "../../test";

jest.mock("uuid", () => ({
  v4: jest.fn(() => "be3a62d1-24a0-401c-96dd-3aff95356811"),
}));

MockDate.set("2020-01-01 08:00:00.000");

describe("assertAuthorizationIsNotExpired", () => {
  test("should succeed when date is before", () => {
    expect(assertAuthorizationIsNotExpired(getTestAuthorization({}))).toMatchSnapshot();
  });

  test("should throw error when date is after", () => {
    expect(() =>
      assertAuthorizationIsNotExpired(
        getTestAuthorization({
          expires: new Date("1999-01-01 01:00:00.000"),
        }),
      ),
    ).toThrow(expect.any(ExpiredAuthorizationError));
  });
});
