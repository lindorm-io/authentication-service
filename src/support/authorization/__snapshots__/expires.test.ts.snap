// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assertAuthorizationIsNotExpired should succeed when date is before 1`] = `undefined`;
